src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+    /* 0x51 */ SCENE_SPOT00,
+    /* 0x52 */ SCENE_SPOT01,
+    /* 0x53 */ SCENE_SPOT02,
+    /* 0x54 */ SCENE_SPOT03,
+    /* 0x55 */ SCENE_SPOT04,
+    /* 0x56 */ SCENE_SPOT05,
+    /* 0x57 */ SCENE_SPOT06,
+    /* 0x58 */ SCENE_SPOT07,
+    /* 0x59 */ SCENE_SPOT08,
+    /* 0x5A */ SCENE_SPOT09,
+    /* 0x5B */ SCENE_SPOT10,
+    /* 0x5C */ SCENE_SPOT11,
+    /* 0x5D */ SCENE_SPOT12,
+    /* 0x5E */ SCENE_SPOT13,
+    /* 0x5F */ SCENE_SPOT15,
+    /* 0x60 */ SCENE_SPOT16,
+    /* 0x61 */ SCENE_SPOT17,
+    /* 0x62 */ SCENE_SPOT18,
+    /* 0x63 */ SCENE_SPOT20,
+};
+
+// Is current scene a SPOT scene
+s32 func_8003BF18(GlobalContext* globalCtx) {
+    s16* i;
+
+    for (i = spotScenes; i < spotScenes + ARRAY_COUNT(spotScenes); i++) {
+        if (globalCtx->sceneNum == *i) {
+            return true;
+        }
+    }
+    return false;
+}
+
+struct BgChkMem {
+    s16 sceneId;
+    u32 size;
+};
 
-u32 func_80041A28(CollisionContext* a0, u32 a1, u32 a2) {
-    u32 uVar1;
+struct BgChkMem D_80119E54[8] = { { SCENE_SPOT00, 0xB798 },     { SCENE_GANON_FINAL, 0x78C8 },
+                                  { SCENE_GANON_DEMO, 0x70C8 }, { SCENE_JYASINBOSS, 0xACC8 },
+                                  { SCENE_KENJYANOMA, 0x70C8 }, { SCENE_JYASINZOU, 0x16CC8 },
+                                  { SCENE_HIDAN, 0x198C8 },     { SCENE_GANON_BOSS, 0x84C8 } };
 
-    uVar1 = func_800419B0(a0, a1, a2, 0);
-    return uVar1 & 0xff;
+// Get custom scene memSize
+s32 func_8003BF5C(s32 sceneId, u32* memSize) {
+    s32 i;
+    for (i = 0; i < 8; i++) {
+        if (sceneId == D_80119E54[i].sceneId) {
+            *memSize = D_80119E54[i].size;
+            return true;
+        }
+    }
+    return false;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041A4C.s")
+// Compute division unit sizes for scene mesh lookup, for a single dimension
+void func_8003BFF4(f32 min, s32 subdivisions, f32* max, f32* unitSize, f32* inverseUnitSize) {
+    f32 tempUnitSize;
+    f32 length;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041B24.s")
+    length = (*max - min);
+    tempUnitSize = (s32)(length / subdivisions) + 1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041B80.s")
+    *unitSize = tempUnitSize;
+    *unitSize = (tempUnitSize < 150.0f) ? 150.0f : *unitSize;
+    *inverseUnitSize = 1.0f / *unitSize;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041C10.s")
+    *max = *unitSize * subdivisions + min;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80041C98.s")
+struct BgChkMemDungeon {
+    s16 sceneId;
+    Vec3s subdivisions;
+    s32 nodeListMax; // if -1, dynamically compute max nodes
+};
+
+struct BgChkMemDungeon D_80119E94[2] = { { SCENE_HAKADAN, { 23, 7, 14 }, -1 }, { SCENE_BMORI1, { 38, 1, 38 }, -1 } };
+
+// Allocate CollisionContext
+void func_8003C078(CollisionContext* colCtx, GlobalContext* globalCtx, CollisionHeader* colHeader) {
+    u32 tblMax;
```

Comment:
is the collision context its own system? if so then CollisionContext_Init or something
or if not BgCheck_InitCollisionContext or something along those lines